dfastmi/io/map_file.py

```diff
@@ -0,0 +1,347 @@
+
+This library is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation version 2.1.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, see <http://www.gnu.org/licenses/>.
+
+contact: delft3d.support@deltares.nl
+Stichting Deltares
+P.O. Box 177
+2600 MH Delft, The Netherlands
+
+All indications and logos of, and references to, "Delft3D" and "Deltares"
+are registered trademarks of Stichting Deltares, and remain the property of
+Stichting Deltares. All rights reserved.
+
+INFORMATION
+This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
+"""
+from pathlib import Path
+from typing import List, Optional
+import numpy as np
+import numpy.ma as ma
+import netCDF4 as nc
+    
+class MapFile:
+    def __init__(self, map_file: Path):
+        """Initializes a new instance of the 'MapFile' class for the provided map file.
+
+        Arguments
+        ---------
+            map_file : Path
+                The path to the map file.
+        """
+        self._map_file = map_file
+        self._mesh2d_name = None
+        self._face_dimension_name = None
+    
+    @property
+    def node_x_coordinates(self) -> np.ndarray:
+        """Get the x-coordinates of the nodes.
+
+        Returns
+        -------
+        numpy.ndarray
+            Array with shape (N,) where N is the number of nodes.
+        """
+        return self._get_node_coordinate_data(["projection_x_coordinate", "longitude"])
+    
+    @property
+    def node_y_coordinates(self) -> np.ndarray:
+        """Get the y-coordinates of the nodes.
+
+        Returns
+        -------
+        numpy.ndarray
+            Array with shape (N,) where N is the number of nodes.
+        """
+        return self._get_node_coordinate_data(["projection_y_coordinate", "latitude"])
+    
+    @property
+    def face_node_connectivity(self) -> ma.masked_array:
+        """Get the face-node connectivity from the 2d mesh.
+
+        Returns
+        -------
+        ma.masked_array
+            Array with shape (N,M) where N is the number of faces and M the maximum number of nodes per face.
+            If not all the faces have the same number of nodes, a boolean mask is provided with shape (N,M) 
+            where each True value indicates a fill value.
+        """
+        with nc.Dataset(self._map_file) as rootgrp:
+            mesh2d = rootgrp.variables[self.mesh2d_name]
+            var_name = mesh2d.getncattr("face_node_connectivity")
+            var = rootgrp.variables[var_name]
+            data = var[...] - MapFile._get_start_index(var)
```

Comment:
is there a specific reason for the Ellipsis, cant you just state 'var' here to get the whole list?

Reply:
var is a netcdf4 variable, with [...] the variable array is indexed to a numpy array.